code/modules/mob/living/simple_animal/hostile/oldzombie.dm

```diff
@@ -0,0 +1,506 @@
+/mob/living/simple_animal/hostile/oldzombie
+	name = "zombie"
+	desc = "A horrifying, shambling carcass of what once resembled a human being."
+	icon = 'icons/mob/human.dmi'
+	icon_state = "zombie_s"
+	icon_living = "zombie_s"
+	icon_dead = "zombie_dead"
+	turns_per_move = 5
+	speak_emote = list("groans")
+	emote_see = list("groans")
+	a_intent = "harm"
+	maxHealth = 100
+	health = 100
+	speed = 3
+	harm_intent_damage = 2
+	melee_damage_lower = 10
+	melee_damage_upper = 15
+	attacktext = "claws"
+	attack_sound = list('sound/weapons/claw_strike1.ogg','sound/weapons/claw_strike2.ogg','sound/weapons/claw_strike3.ogg')
+	atmos_requirements = list("min_oxy" = 0, "max_oxy" = 0, "min_tox" = 0, "max_tox" = 0, "min_co2" = 0, "max_co2" = 5, "min_n2" = 0, "max_n2" = 0)
+	minbodytemp = 0
+	maxbodytemp = 370
+	unsuitable_atmos_damage = 10
+	environment_smash = 1
+	robust_searching = 1
+	stat_attack = 2
+	faction = list("zombie")
+	var/mob/living/carbon/human/stored_corpse = null
+	butcher_results = list(/obj/item/weapon/reagent_containers/food/snacks/meat/slab/human/mutant/zombie = 3)
+	see_invisible = SEE_INVISIBLE_MINIMUM
+	see_in_dark = 8
+	layer = MOB_LAYER - 0.1
+	var/removingairlock = 0
+	var/can_possess = 1
+	var/startinfected = 1 //to check if they have ever been infected before.
+	var/spam_flag_z = 0
+	var/numinfected = 5
+	var/selfrevive = 0
+	var/forcedoor = 0
+	var/airlocktime = 200
+	var/superform = 0
+	var/superformtime = 100
+	languages = ZOMBIE
+
+/mob/living/simple_animal/hostile/oldzombie/AttackingTarget()
+	..()
+	var/mob/living/L = target
+	if(istype(L, /mob))
+		if(ishuman(L))
+			var/mob/living/carbon/human/H = L
+			if(H.infected == 1)
+				src << "<span class='userdanger'>[L] is already infected! Give them time!</span>"
+			else
+				var/infectchance = rand(1,3)
+				if(infectchance == 2)
+					if(H.startinfected == 1)
```

Comment:
This value is 1 by default, meaning if you infect anyone else you'll always get 50 health regenerated and 3 extra points, on top of 20 health regenerated and 1 extra point